pydantic/_internal/_validate_call.py

```diff
@@ -37,6 +37,11 @@ def __init__(self, function: Callable[..., Any], config: ConfigDict | None, vali
             self.__module__ = function.__module__
 
         namespace = _typing_extra.add_module_globals(function, None)
+        # TODO: this is a bit of a hack, we should probably have a better way to handle this
+        # specifically, we shouldn't be pumping the namespace full of type_params
+        # when we take namespace and type_params arguments in eval_type_backport
+        type_params = getattr(schema_type, '__type_params__', ())
+        namespace.update({param.__name__: param for param in type_params})
```

Comment:
```suggestion
        namespace = {param.__name__: param for param in type_params} | namespace
```

This gives `type_params` lower precedence which is important

Reply:
Ah yeah, but the `|` syntax isn't supported in 3.8

Reply:
But I'll do dict unpacking to preserve that